pkg/api/http.go

```diff
@@ -337,6 +335,13 @@ func ParseQueryRangeRequest(r *http.Request) (*tempopb.QueryRangeRequest, error)
 	req.Query = r.Form.Get("query")
 	req.QueryMode = r.Form.Get(QueryModeKey)
 
+	// if we don't have a query then let's attempt to parse them out of the url
+	//  Grafana posts these params as form data, but BuildQueryRangeRequest below puts them in the url
+	if len(req.Query) == 0 && len(req.QueryMode) == 0 {
+		req.Query = r.Form.Get(urlParamQuery)
```

Comment:
It looks like we are checking for both `query` and `q`.  Since we are dropping Prom support I was hoping we could drop one, but unfortunately it looks like we are using `query` for query_range and `q` for all other search endpoints.    Probably ok to leave both then and we can consolidate starting with Grafana.

Either way, I think this area could be simplifed to make the handling of the query and mode separate, it looks like it calls `r.Form.Get(QueryModeKey)` twice?

Reply:
> It looks like we are checking for both query and q. Since we are dropping Prom support I was hoping we could drop one, but unfortunately it looks like we are using query for query_range and q for all other search endpoints. Probably ok to leave both then and we can consolidate starting with Grafana.

I added this b/c I figured Grafana was using this form post and not url params. Since traceql metrics are already in explore I figured we had to keep it, but let me run a test.

> Either way, I think this area could be simplifed to make the handling of the query and mode separate, it looks like it calls r.Form.Get(QueryModeKey) twice?

Agree there's a bug here. Those internal calls should be attempting to pull from the url. It does have me wondering how it's working at all :). Will dig in.

Reply:
Ok, I see. `ParseForm` pulls in the url parameters to the form which is why it works. I've updated the code and added some comments to clearly always pull from the url. This works with Grafana/CLI.